### PR TITLE
PSR-602 shares total income page

### PR DESCRIPTION
**Pull request by narendrapaduchuri**

### CODE REVIEW

File: app/navigation/nonsipp/SharesNavigator.scala [hunk 1]
```diff
@@ -161,7 +161,21 @@ object SharesNavigator extends JourneyNavigator {
       controllers.nonsipp.shares.routes.SharesIndependentValuationController.onPageLoad(srn, index, NormalMode)
 
     case SharesIndependentValuationPage(srn, index) =>
+      userAnswers.get(WhyDoesSchemeHoldSharesPage(srn, index)) match {
+        case Some(Acquisition) =>
+          controllers.routes.UnauthorisedController.onPageLoad()
```
Comment (ccatosdevelopment) [inline app/navigation/nonsipp/SharesNavigator.scala:166]: If this is for the conditional page, then within this `Acquisition` branch I think there also needs to be a check that `TypeOfShares` is `SponsoringEmployer`. And if it's `Unquoted` or `ConnectedParty`, then it navigates `SharesTotalIncomeController` instead.

File: app/navigation/nonsipp/SharesNavigator.scala [hunk 2]
```diff
@@ -161,7 +161,21 @@ object SharesNavigator extends JourneyNavigator {
...
+        case _ =>
```
Comment (ccatosdevelopment) [inline app/navigation/nonsipp/SharesNavigator.scala:167]: If it's a `Contribution` or `Transfer`, then I think the navigation should go to `SharesTotalIncomeController` in any case, regardless of whether `TypeOfShares` is `SponsoringEmployer`, `Unquoted`, or `ConnectedParty`.

File: app/navigation/nonsipp/SharesNavigator.scala (hunk 1 above)
Comment (BabuRupoll7623) [reply app/navigation/nonsipp/SharesNavigator.scala:166]: changing now

File: app/navigation/nonsipp/SharesNavigator.scala (hunk 2 above)
Comment (BabuRupoll7623) [reply app/navigation/nonsipp/SharesNavigator.scala:167]: we are discussing same here, changing now

**Review by ccatosdevelopment — APPROVED** [review]

**Pull request merged**